Rename base service version property to serviceVersion

mkistler opened the pull request:
This PR changes the name of the internal `version` property of BaseService to `serviceVersion`.  I believe this is purely an internal change.  All tests pass.

The purpose of this change is to lay the foundation for using 'version' as the parameter name for the "version date" that is passed in to some services.

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `npm test` passes (tip: `npm run autofix` can correct most style issues)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/watson-developer-cloud/node-sdk?pullRequest=628) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/watson-developer-cloud/node-sdk?pullRequest=628) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/watson-developer-cloud/node-sdk?pullRequest=628) it.</sub>

germanattanasio commented on index.ts:
```diff
@@ -65,16 +65,16 @@ Object.keys(servicesByVersion).forEach(serviceName => {
       }
 
-      const Service = servicesByVersion[serviceName][options.version];
+      const Service = servicesByVersion[serviceName][options.serviceVersion];
```
This is a breaking change 🔥 

Users should still be able to use

```js
var conversationV1 = watson.conversation({
  version: 'v1',
  version_date: '2017-05-26'
});
```

mkistler replied:
Okay ... I see now.  We documented this style in past releases.  It's good to see that we have removed the doc on this.  I will think about how we can handle that.

mkistler commented:
I've reworked this change to support the "old style" of service instantiation, but I added a deprecation warning to discourage folks from using it.  I reverted all the changes in the test and examples directory, and all (unchanged) tests pass with this new implementation.

germanattanasio commented on index.ts:
```diff
@@ -78,7 +89,10 @@ Object.keys(servicesByVersion).forEach(serviceName => {
       }
 
-      return new Service(options);
+      const _options = extend({}, options);
```
Did you try with: 
```js
return new Service({
  ...options,
  serviceVersion: options.version,
  version: options.version_date,
});
```

germanattanasio commented:
This was released as `beta` in `3.0.7`
You can install it by running:
```js
npm install watson-developer-cloud@beta
```

glennrfisher commented:
Doesn't Natural Language Understanding still need to be updated [here][1] and [here][2]?

As an aside, it can be tricky to follow changes when they're rebased. I think the general recommendation is not to rebase code that's been pushed. (Although there are some nuances, like rebasing a PR into master instead of merging it.)

[1]: https://github.com/watson-developer-cloud/node-sdk/blob/34ae1e74ab55a76a2d2e2665fb75aeefe5b85090/natural-language-understanding/v1.ts#L29

[2]: https://github.com/watson-developer-cloud/node-sdk/blob/34ae1e74ab55a76a2d2e2665fb75aeefe5b85090/natural-language-understanding/v1.ts#L210

[3]: https://www.atlassian.com/git/articles/git-team-workflows-merge-or-rebase

mkistler commented:
@glennrfisher I believe you are right.  Thanks for catching these.  I guess the tests don't exercise the old constructor for NLU.  But I will fix.
